File: Data Structures & Algorithms/find-target-in-rotated-sorted-array/submission-2.java
class Solution {
    public int search(int[] nums, int target) {

        int len = nums.length;
        int start =0, end = len-1, mid =0;

        while(start<=end){
            mid=(start+end)/2;
            System.out.println("mid : "+nums[mid]);
            if(nums[mid]>target){
                if(nums[start]>target && nums[start]<=nums[mid]){
                    start = mid+1;
                }else{
                    end = mid-1;
                }
            }else if(nums[mid]<target){
                if(nums[end]<target && nums[end]>=nums[mid]){
                    end = mid-1;
                }else{
                    start = mid+1;
                }

            }else{
                return mid;
            }

        }
        return -1;
    }
}

File: Data Structures & Algorithms/two-integer-sum-ii/submission-2.js
class Solution {
    /**
     * @param {number[]} numbers
     * @param {number} target
     * @return {number[]}
     */
    twoSum(numbers, target) {
        let start=0;
        let end = numbers.length-1;
        let current = 0;
        while(start<end){
            current = numbers.at(start) + numbers.at(end);
            if(current < target){
                start++;
            }else if(current > target){
                end--;
            }else{
                return [start+1,end+1];
            }
        }
        return [1,2];
    }
}
